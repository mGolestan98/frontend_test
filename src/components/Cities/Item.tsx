import React from "react";
import styled from "styled-components";
import { ICity } from "../../types";

const StyledItem = styled.li``;

type PropsType = { city: ICity };

const Item = ({ city }: PropsType) => <StyledItem>{city.city}</StyledItem>;

export default Item;
