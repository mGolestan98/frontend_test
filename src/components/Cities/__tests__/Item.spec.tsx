import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Item from "../Item";

describe("City", () => {
  describe("Snaphosts", () => {
    function getTreeComponent(props = {}) {
      return renderer
        .create(
          <Item
            city={{
              city: "New York",
              growth_from_2000_to_2013: "4.8%",
              latitude: 40.7127837,
              longitude: -74.0059413,
              population: "8405837",
              rank: "1",
              state: "New York"
            }}
            {...props}
          />
        )
        .toJSON();
    }
    it("renders", () => {
      expect(getTreeComponent()).toMatchSnapshot();
    });
  });
});
