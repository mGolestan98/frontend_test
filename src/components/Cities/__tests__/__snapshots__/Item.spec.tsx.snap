// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Snaphosts renders 1`] = `
<li
  className="sc-Axmtr jSpriw"
  onClick={[Function]}
>
  <div
    className="sc-AxirZ cVwUnI"
  >
    <h2
      className="sc-AxhCb cOqwkx"
    >
      New York
    </h2>
    <p
      className="sc-AxhUy hdigZc"
    >
      Rank: 
      1
    </p>
  </div>
  <div
    className="sc-AxirZ deKsjO"
  >
    <p
      className="sc-AxhUy jeEzEq"
    >
      Population: 
      8,405,837
    </p>
    <div
      className="sc-AxmLO fRRgPH"
    >
      <p
        className="sc-AxhUy jeEzEq"
      >
        Growth: 
        4.8%
      </p>
      <span
        className="sc-AxgMl kUaHlF"
      />
    </div>
  </div>
</li>
`;
